apps/src/templates/sectionProgressV2/ProgressTableV2.jsx

```diff
@@ -3,38 +3,93 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import {studentShape} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
 import StudentColumn from './StudentColumn';
-import ProgressTableHeader from './ProgressTableHeader';
-import ProgressDataV2 from './ProgressDataV2';
 import styles from './progress-table-v2.module.scss';
 import stringKeyComparator from '@cdo/apps/util/stringKeyComparator';
 import {getCurrentUnitData} from '../sectionProgress/sectionProgressRedux';
 import {unitDataPropType} from '../sectionProgress/sectionProgressConstants';
+import ExpandedProgressDataColumn from './ExpandedProgressDataColumn';
+import LessonsProgressDataColumn from './LessonsProgressDataColumn';
 
 export function ProgressTableV2({
   isSortedByFamilyName,
   sectionId,
   students,
   unitData,
+  expandedLessonIds,
+  setExpandedLessons,
 }) {
   const sortedStudents = React.useMemo(() => {
     return isSortedByFamilyName
       ? students.sort(stringKeyComparator(['familyName', 'name']))
       : students.sort(stringKeyComparator(['name', 'familyName']));
   }, [students, isSortedByFamilyName]);
 
+  const renderedColumns = React.useMemo(() => {
+    const lessons = unitData?.lessons;
+    if (lessons === undefined || lessons.length === 0) {
+      return [];
+    }
+
+    let currentGroup = [];
+    let columns = [];
+    // Groups adjacent unexpanded lessons together into `LessonsProgressDataColumn`s
+    // Expanded lessons are not grouped with other lessons
+    for (const lesson of lessons) {
+      if (expandedLessonIds.includes(lesson.id)) {
+        if (currentGroup.length > 0) {
```

Comment:
I'm curious if there's an opportunity to simplify the logic by rendering each unexpanded lesson as its own component rather than grouping them together like this. Then it might just be a matter of

```js
const columns = lessons.map(lesson => expandedLessonIds.includes(lesson.id)
  ? (<LessonProgressDataColumn... />)
  : (<ExpandedProgressDataColumn... />)
);
```